### ISSUE TITLE
Feature: allow to combine @SpyK with @InjectMockKs

### ISSUE THREAD

**Issue by vadim-shb** [mockk/mockk#315]
As I can see, the following syntax doesn't work:
```kotlin
    @RelaxedMockK
    private lateinit var someDao: SomeDao

    @InjectMockKs
    @SpyK
    private lateinit var underTest: UnderTest
```
I think it would be great if it's become an equivalent of the next snippet:

```kotlin
    @RelaxedMockK
    private lateinit var someDao: SomeDao

    @InjectMockKs
    private lateinit var underTest: UnderTest

    @BeforeEach
    fun setup() {
        underTest = spyk(underTest)
    }
```
So we'll be able to spy/redefine some out of test methods in the tested class.

**Comment by Quicksteve** [comment]
This would be very nice to have.

**Comment by IlyaEremin** [comment]
Any updates?

**Comment by feltrix** [comment]
Any updates?

**Comment by Raibaz** [comment]
This was fixed by #529 and will be in the next release.
